src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerObserver.cs

```diff
@@ -38,9 +40,30 @@ public Task OpenAsync(ChangeFeedObserverContext context)
             return Task.CompletedTask;
         }
 
-        public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+        public async Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
         {
-            return this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
+            int retries = 0;
+
+            while (true)
+            {
+                var result = await this.executor.TryExecuteAsync(new TriggeredFunctionData() { TriggerValue = docs }, CancellationToken.None);
+
+                if (result.Succeeded)
+                {
+                    // notification processed successfully
+                    return;
+                }
+
+                if (retryCount != -1 && retries >= retryCount)
```

Comment:
If the default value is 0, why are you comparing against -1? Wouldn't this check be: retries == retryCount ?

If the value is 0, it would return on the first error (no retry). If the value is higher it would increment the local variable until it matches the `retryCount` value and also stop.

Reply:
-1 means "infinite retries" as documented on the attribute, so I need to ignore the count of retries in that case

Reply:
Also, I could do "[inf check] && retries == retryCount" instead of "[inf check] && retries >= retryCount" as you suggested, but I used the >= just as sort of failsafe even though the value should never increment more than once per loop.